elytron/WFLY-10810_Elytron_JASPI_Servlet_Profile_Support.adoc

```diff
@@ -0,0 +1,160 @@
+
+=====  CallerPrincipalCallback
+
+The overall intent of this behaviour is to influence the resulting identity for the authentication process.
+* If an identity is already cached and this callback provides a new caller principal we will handle this as a runAs event, in integrated moded we will also perform an authorization check to ensure the cached identity has the appropriate permissions to run as this new identity.
+* If an identity is already cached and the callback does not specify a new caller principal this is taken as a request to use the current identity, in integrated mode an authorization check is performed to ensure this identity is authroized to login.
+* If there is no pre-cached identity and the callback specifies no new caller principal this has no effect.
+* If there is no cached identity and the callback specifies a caller principal, in non-integrated mode a new ad-hoc identity is created from the domain.  In integrated mode the identity will be loaded from the domain and authorization performed to ensure the identity has been granted the login permission.
+
+===== GroupPrincipalCallback
+
+In addition to the 'current' identity, the `CallbackHandler` also caches a list of groups, each time an instance of this Callback is received the listed groups will be added to this cache.
+
+At the very end of the authentication process when the final `SecurityIdentity` is retrieved from the handler if any groups have been cached these will be used to replace the set of `Roles` on the identity.  This replacement set of roles replaces any which may already exist for that identity, if no groups were specified using this callback then the roles on the identity are not modified.
+
+=== Nice-to-Have Requirements
+
+The steps to deploy an application making use of the WildFly Elytron JASPI integration should be minimal, the minimal steps would be: -
+
+. Association with a WildFly Elytron SecurityDomain to activate the Elytron Web handlers.
+. Dynamic registration of an AuthConfigProvider by the application or a pre-existing definition in the Elytron subsystem. 
+
+=== Non-Requirements
+
+It should be noted that when used in non-integrated mode ad-hoc identities will be created from the referenced `SecurityDomain` instead of being loaded by the `SecurityDomain` this will mean that if these identities are used with other components that rely on reloading an identity such as Batch this will not be possible.
+
+== Implementation Plan
+
+The overall implementation of this will affect four different projects, however no special steps should be required provided they are merged in dependency order: -
+
+. WildFly Elytron
+. Elytron Web
+. WildFly Core
+. WildFly 
+
+== Test Plan
+
```

Comment:
Which scenarios do you plan to add/have you added to Elytron and WildFly Core?

TCK also cover some cases which is mapped by QE test plan (draft).

Reply:
I will add a lit tomorrow of the test coverage currently planned, we can then discuss if we want to get any additional scenarios added.